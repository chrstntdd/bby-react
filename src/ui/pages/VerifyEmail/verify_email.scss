@import '$homeDir/styles/rwd-helpers.scss';
@import '$homeDir/styles/colors.scss';

#confirm-email-wrapper {
  min-height: 100vh;
  width: 100vw;
  #confirm-email-card {
    @media (min-width: $bp-xxsmall) {
      width: between(300px, 560px, $bp-xxsmall, $bp-xlarge);
    }
    margin-top: 6rem;
    border-top: 0.3rem solid $BESTBUY_BLUE;
    padding: 0;
    width: between(300px, 660px, $bp-xxsmall, $bp-xlarge);
    padding: between(0, 16px, $bp-xxsmall, $bp-xlarge);
  }
}
