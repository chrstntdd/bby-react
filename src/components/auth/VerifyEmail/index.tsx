import './verify_email.scss';
import 'url-search-params-polyfill';

import * as React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import { confirmEmail } from '../../../state/actions';

export class VerifyEmail extends React.Component {
  componentDidMount() {
    const search = this.props.location.search;
    const params = new URLSearchParams(search);
    const token = params.get('token');
    this.props.confirmEmail(token);
  }

  redirectToDash() {
    this.props.history.push('/dashboard');
  }
  render() {
    return (
      <section id="confirm-email-wrapper">
        <div id="confirm-email-card">
          <h1>Thank you for registering your account!</h1>
          <button onClick={() => this.redirectToDash()}>Get started</button>
        </div>
      </section>
    );
  }
}

const mapStateToProps = state => ({
  //
});

export default withRouter(connect(mapStateToProps, { confirmEmail })(VerifyEmail));
