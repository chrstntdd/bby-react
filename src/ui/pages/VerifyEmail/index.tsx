import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import { confirmEmail } from '@/state/actions';

import './verify_email.scss';

export class VerifyEmail extends Component {
  componentDidMount() {
    /* USE WINDOW.LOCATION.SEARCH */
    const { search } = window.location;
    console.log(search);
    // this.props.confirmEmail(token);
  }

  redirectToDash() {
    this.props.history.push('/dashboard');
  }
  render() {
    return (
      <section id="confirm-email-wrapper">
        <div id="confirm-email-card">
          <h1>Thank you for registering your account!</h1>
          <button onClick={() => this.redirectToDash()}>Get started</button>
        </div>
      </section>
    );
  }
}

export default withRouter(connect(null, { confirmEmail })(VerifyEmail));
