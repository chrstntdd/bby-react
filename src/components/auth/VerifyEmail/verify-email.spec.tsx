import * as React from 'react';
import { shallow } from 'enzyme';

import VerifyEmail from './';

describe('<VerifyEmail/>', () => {
  it('should render without crashing', () => {
    const wrapper = shallow(<VerifyEmail />);

    expect(wrapper).toMatchSnapshot();
  });
});
