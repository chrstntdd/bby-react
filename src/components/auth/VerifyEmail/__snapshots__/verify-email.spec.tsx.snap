// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VerifyEmail/> should render without crashing 1`] = `
<Route
  render={[Function]}
/>
`;
